app/knights/knight.py

```diff
@@ -0,0 +1,35 @@
+from .equipment import Armour, Weapon, Potion
+
+
+class Knight:
+    """Represents a knight with stats and equipment."""
+
+    def __init__(
+        self, name: str, power: int, hp: int, armour: str, weapon: str,
+        potion: str = None
```

Comment:
The parameters `armour`, `weapon`, and `potion` are expected to be dictionaries (or lists of dictionaries for `armour`). Ensure that the input data matches this expectation to avoid runtime errors.